ENIGMAsystem/SHELL/Widget_Systems/Win32/dialogs.cpp

```diff
@@ -73,9 +73,18 @@ void show_error(string errortext, const bool fatal)
 }
 
 namespace enigma {
-  extern HINSTANCE hInstance;
-  extern HWND hWnd;
-  HWND infore;
+
+extern HINSTANCE hInstance;
+extern HWND hWnd;
+HWND infore;
+
+}
+
+static inline string remove_slash(string dir) {
+  if (!dir.empty() && *dir.rbegin() == '\\') {
```

Comment:
use .back() to get last char in a std::string not deference the reverse iterator. Also this doen't deal with forward slashes

Reply:
The way things are set up it is not possible for it to end in a forward slash unless of course if run from wine but really people should be make games native for Linux instead. I'll humor you and make the change anyway.

Reply:
Also used rbegin because thats what Josh used when he wrote the add_slash() function. You can complain to him about that one. All of these things i did you called "tasteless" werent even my decision or idea to begin with. Im just doing what people tell me.

Reply:
Yes, you do as you're told cause you aren't intelligent enough to make a pr as simple as this on your own. If you ever manager to learn to read above a 2nd grade lvl there's a a reference on std::string functions here https://en.cppreference.com/w/cpp/string/basic_string

Reply:
@gfundies the real issue here is you all are making me do changes that aren't needed and you all disagree on how it should be done so maybe it is you that is living in second grade still and can't do anything without Josh or Robert needing to correct it or vise versa.